subprojects/plugins/src/integTest/groovy/org/gradle/java/JavaLibraryFeatureCompilationIntegrationTest.groovy

```diff
@@ -564,7 +564,7 @@ class JavaLibraryFeatureCompilationIntegrationTest extends AbstractIntegrationSp
                 [apiElements, runtimeElements].each {
                     assert it.canBeConsumed == true
                     assert it.canBeResolved == false
-                    assert it.canBeDeclaredAgainst == true
```

Comment:
Elements should not have been declarable - I believe this was overly permissive and shouldn't be the case.

Reply:
We can't just change the usage here without a deprecation, though. We need to leave this as declarable, while also deprecating it for declarable.